PurchasesCoreSwift/Misc/SystemInfo.swift

```diff
@@ -0,0 +1,157 @@
+        #elseif os(watchOS)
+            return WKInterfaceDevice.current().identifierForVendor?.uuidString
+        #else
+            return nil
+        #endif
+    }
+
+    private static var defaultServerHostURL: URL {
+        return URL(string: defaultServerHostName)!
+    }
+
+    @objc public static var serverHostURL: URL {
+        return self.proxyURL ?? Self.defaultServerHostURL
+    }
+
+    private static var privateProxyURL: URL?
+    @objc public static var proxyURL: URL? {
+        get {
+            return privateProxyURL
+        }
+        set {
+            privateProxyURL = newValue
+            if let privateProxyURLString = newValue?.absoluteString {
+                Logger.info(Strings.configure.configuring_purchases_proxy_url_set
+                                .replacingOccurrences(of: "%@", with: privateProxyURLString))
+            }
+        }
+    }
+
+    @objc required public init(platformFlavor: String?, platformFlavorVersion: String?, finishTransactions: Bool) throws {
+        self.platformFlavor = platformFlavor ?? "native"
+        self.platformFlavorVersion = platformFlavorVersion
+
+        if (platformFlavor == nil && platformFlavorVersion != nil) ||
+            (platformFlavor != nil && platformFlavorVersion == nil) {
+            Logger.error("RCSystemInfo initialized with non-matching platform flavor and platform flavor versions!")
+            throw SystemInfoError.invalidInitializationData
+        }
+
+        self.finishTransactions = finishTransactions
+    }
+
+    @objc open func isApplicationBackgrounded(completion: @escaping (Bool) -> Void) {
+        DispatchQueue.main.async {
+            completion(self.isApplicationBackgrounded)
+        }
+    }
+
+    @objc open func isOperatingSystemAtLeastVersion(_ version: OperatingSystemVersion) -> Bool {
+        return ProcessInfo.processInfo.isOperatingSystemAtLeast(version)
+    }
+
+    private var isApplicationBackgrounded: Bool {
+    #if os(iOS) || (targetEnvironment(simulator) && os(iOS))
+        return self.isApplicationBackgroundedIOS
+    #elseif os(tvOS)
+        return UIApplication.shared.applicationState == UIApplication.State.background
+    #elseif os(macOS)
+        return false
+    #elseif os(watchOS)
+        return  WKExtension.shared().applicationState == WKApplicationState.background
+    #endif
+    }
+
+    #if os(iOS)
+    // iOS App extensions can't access UIApplication.sharedApplication, and will fail to compile if any calls to
+    // it are made. There are no pre-processor macros available to check if the code is running in an app extension,
+    // so we check if we're running in an app extension at runtime, and if not, we use KVC to call sharedApplication.
+    private var isApplicationBackgroundedIOS: Bool {
+        if self.isAppExtension {
+            return true
+        }
+
+        return UIApplication.shared.applicationState == UIApplication.State.background
```

Comment:
this line breaks compilation for app extensions. `shared` isn't available for them, and it will fail at compilation time, which is why we were doing it this way before: https://github.com/RevenueCat/purchases-ios/blob/main/Purchases/Misc/RCSystemInfo.m#L139